hooks/module.go

```diff
@@ -0,0 +1,6 @@
+package hooks
+
+type Module interface {
+	Name() string
```

Comment:
Does the Name need to be a part of the interface? .. or can we link it from the Builders map?